clients/context-registry/src/ContextRegistryClient.ts

```diff
@@ -115,122 +106,96 @@ export class Project extends CommonAssetProjectContext {
    * @param requestContext The client request context
-   * @param queryOptions Query options. Use the mapped EC property names in the query strings and not the TypeScript property names.
-   * @returns Resolves to the found asset. Rejects if no assets, or more than one asset is found.
+   * @param arg Options for paging and/or searching
+   * @returns Array of iTwins, may be empty
    */
-  public async getAsset(requestContext: AuthorizedClientRequestContext, queryOptions?: RequestQueryOptions): Promise<Asset> {
-    requestContext.enter();
-    const assets: Asset[] = await this.getAssets(requestContext, queryOptions);
-    requestContext.enter();
-    if (assets.length === 0)
-      throw new Error("Could not find an asset with the specified criteria that the user has access to");
-    else if (assets.length > 1)
-      throw new Error("More than one asset found with the specified criteria");
-
-    return assets[0];
+  public async getAll(requestContext: AuthorizedClientRequestContext, arg?: ITwinQueryArg): Promise<ITwin[]> {
+    const queryOptions: RequestQueryOptions = {
+      $top: arg?.pagination?.top,
+      $skip: arg?.pagination?.skip,
+    };
+
+    if (arg?.search) {
+      if (arg.search.exactMatch)
+        queryOptions.$filter = `${arg.search.propertyName}+eq+'${arg.search.searchString}'`;
+      else
+        queryOptions.$filter = `${arg.search.propertyName}+like+'${arg.search.searchString}'`;
+    }
+
+    return this.getByQuery(requestContext, queryOptions);
   }
 
-  /** Gets the iTwin asset contexts that are accessible to the authorized user.
+  /** Gets all iTwins (projects or assets) using the given query options
    * @param requestContext The client request context
-   * @param queryOptions Query options. Use the mapped EC property names in the query strings and not the TypeScript property names.
-   * @returns Resolves to an array of assets.
+   * @param queryOptions Use the mapped EC property names in the query strings and not the TypeScript property names.
+   * @returns Array of iTwins meeting the query's requirements
    */
-  public async getAssets(requestContext: AuthorizedClientRequestContext, queryOptions?: RequestQueryOptions): Promise<Asset[]> {
+  private async getByQuery(requestContext: AuthorizedClientRequestContext, queryOptions?: RequestQueryOptions): Promise<ITwin[]> {
     requestContext.enter();
-    return this.getInstances<Asset>(requestContext, Asset, "/Repositories/BentleyCONNECT--Main/ConnectedContext/Asset", queryOptions);
+    const projectQuery = queryOptions;
+    const assetQuery = queryOptions;
+
+    if (queryOptions?.$skip && (projectQuery && assetQuery)) {
+      // Ceiling to skip a project object on Odd skips
+      projectQuery.$skip = queryOptions?.$skip ? Math.ceil(queryOptions.$skip / 2) : undefined;
+
+      // Floor to skip an asset object on Even skips
+      assetQuery.$skip = queryOptions?.$skip ? Math.floor(queryOptions.$skip / 2) : undefined;
+    }
```

Comment:
This is affecting the same object to 2 different references, if you then change a property of any of these object, all the values are changed (so you will always get the `assetQuery.$skip` value to all values, and your values will be devided twice because by the time you do the `assetQuery.$skip`, you already devided the original `$skip` value by 2, if there was one.

Also, the `if` do not need the `(projectQuery && assetQuery)` as if `queryOptions.$skip` is defined, then the two other objects are defined as they are copies of the first one.

Reply:
Also, the overall math of this function seems off unless this is used in a very specific way, given that I do pages of 100,  `$skip / 2` would result in weird behavior; I want to skip the 5000 first itwins, and get 5000 to 5100, so I'd enter `$top=100, $skip=5000`, and I'd get projects 2500 to 2600...

Reply:
And I would actually only get 2500 to 2550 as it then slices the results at max half of the results.

Reply:
I see what you mean about the references. I am working on a smart way to do a deep copy.
The `(projectQuery && assetQuery)` is in the `if` conditional because typescript and the linter struggle with chance of them being undefined. The conditional is required otherwise it won't build since it thinks `projectQuery` or `assetQuery` may be undefined causing an error on their assignment. This is also why writing a deep copy is proving difficult as well.

Reply:
The math is a bit difficult to follow and not what I would like long term. This was the best solution I could come up with the constraint of having to silently merge two different endpoints without the user realizing.

Reply:
@raplemie I am using the spread operator now to copy the top level primitives. This should separate the object references.